sqlmesh/core/node.py

```diff
@@ -33,6 +33,59 @@ class IntervalUnit(str, Enum):
     def is_date_granularity(self) -> bool:
         return self in (IntervalUnit.YEAR, IntervalUnit.MONTH, IntervalUnit.DAY)
 
+    @property
+    def _cron_expr(self) -> str:
+        if self == IntervalUnit.MINUTE:
+            return "* * * * *"
+        if self == IntervalUnit.HOUR:
+            return "0 * * * *"
+        if self == IntervalUnit.DAY:
+            return "0 0 * * *"
+        if self == IntervalUnit.MONTH:
+            return "0 0 1 * *"
+        if self == IntervalUnit.YEAR:
+            return "0 0 1 1 *"
+        return ""
+
+    def croniter(self, value: TimeLike) -> CroniterCache:
+        return CroniterCache(self._cron_expr, value)
```

Comment:
this is getting created every time you call it, maybe it's ok since it's not expensive to create and the cache still hits?

Reply:
yeah, it doesn't appear that we call it in the loop. The only place where it's called in the loop it's used as a fallback and can only happen on the last iteration: https://github.com/TobikoData/sqlmesh/blob/e478491aaeb397ae3fc8881c593d12f3d243fe90/sqlmesh/core/snapshot/definition.py#L696